File: backend/src/main/java/todoktodok/backend/member/application/dto/response/ProfileResponse.java
package todoktodok.backend.member.application.dto.response;

import todoktodok.backend.member.domain.Member;

public record ProfileResponse(
        Long memberId,
        String nickname,
        String profileMessage,
        String profileImage
) {

    public ProfileResponse(final Member member) {
        this(
                member.getId(),
                member.getNickname(),
                member.getProfileMessage(),
                member.getProfileImage()
        );
    }
}

File: android/TodokTodok/app/src/main/java/com/team/todoktodok/data/network/response/ProfileResponse.kt
package com.team.todoktodok.data.network.response

import com.team.domain.model.member.Profile
import kotlinx.serialization.Serializable

@Serializable
data class ProfileResponse(
    val memberId: Long,
    val nickname: String,
    val profileMessage: String,
    val profileImage: String,
) {
    fun toDomain(): Profile =
        Profile(
            memberId,
            nickname,
            profileMessage,
            profileImage,
        )
}
